packages/components/select/src/select.js

```diff
@@ -67,23 +81,18 @@ export class TSSelect extends TSElement {
 		this.opened = true;
 	}
 
-	firstUpdated() {
-		window.addEventListener('resize', this.resizeListener);
-		window.addEventListener('mousedown', this.clickOutsideListener);
-	}
-
 	disconnectedCallback() {
 		window.removeEventListener('resize', this.resizeListener);
-		window.removeEventListener('mousedown', this.clickOutsideListener);
+		window.removeEventListener('click', this.clickOutsideListener);
```

Comment:
I can't find where you add this listener.

Reply:
oops, this is actually a way to fix this issue:
when user clicks on the button outside the dropdown, this listener get triggered first and close the dropdown. Because of that the toggle function will always open the dropdown, because it will be closed at the time of onClick callback gets called.
To fix this I changed `mousedown` to `click` plus I also had to call a `stopPropagation` of the `click` event in button. I update the PR with this fix.